dd-sdk-android-sqldelight/src/main/kotlin/com/datadog/android/sqldelight/SqlDelightExt.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sqldelight
+
+import com.datadog.android.tracing.AndroidTracer
+import com.squareup.sqldelight.Transacter
+import com.squareup.sqldelight.Transacter.Transaction
+import com.squareup.sqldelight.TransactionWithReturn
+import com.squareup.sqldelight.TransactionWithoutReturn
+import io.opentracing.Span
+import io.opentracing.util.GlobalTracer
+
+/**
+ * Starts a [Transaction] and runs [body] in that transaction.
+ * A span will be created around the transaction code and sent to Datadog.
+ *
+ * @param operationName the name of the [Span] created around the coroutine code.
+ * @param noEnclosing in case we want the currently opened transaction to be automatically closed
+ * @param body the code to be executed inside the transaction
+ * @throws IllegalStateException if [noEnclosing] is true and there is already an active
+ *   [Transaction] on this thread.
+ */
+fun <T : Transacter> T.transactionTraced(
+    operationName: String,
+    noEnclosing: Boolean = false,
+    body: TransactionWithoutReturn.() -> Unit
+) {
+    withinSpan(operationName, GlobalTracer.get().activeSpan()) {
+        transaction(noEnclosing = noEnclosing, body = body)
+    }
+}
+
+/**
+ * Starts a [Transaction] and runs [body] in that transaction.
+ * A span will be created around the transaction code and sent to Datadog.
+ *
+ * @param operationName the name of the [Span] created around the coroutine code.
+ * @param noEnclosing in case we want the currently opened transaction to be automatically closed
+ * @param body the code to be executed inside the transaction
+ * @throws IllegalStateException if [noEnclosing] is true and there is already an active
+ *   [Transaction] on this thread.
+ */
+fun <T : Transacter, R> T.transactionTracedWithResult(
+    operationName: String,
+    noEnclosing: Boolean = false,
+    body: TransactionWithReturn<R>.() -> R
+): R {
+    withinSpan(operationName, GlobalTracer.get().activeSpan()) {
+        return transactionWithResult(noEnclosing = noEnclosing, bodyWithReturn = body)
+    }
+}
+
+@Suppress("ThrowingInternalException", "TooGenericExceptionCaught")
+internal inline fun <T : Any?> withinSpan(
+    operationName: String,
+    parentSpan: Span? = null,
+    block: () -> T
```

Comment:
Maybe you could make this block type : `Span.() -> T` so that people can add custom tags to the span if need be.

Reply:
Hmm...that means I should make this one `body: TransactionWithoutReturn.() -> Unit` to something like `body: SpanTransactionWithoutReturn.()->Unit`. I need to see how can I achieve this.